src/include/mpir_request.h

```diff
@@ -609,11 +613,6 @@ MPL_STATIC_INLINE_PREFIX int MPIR_Request_completion_processing_fastpath(MPI_Req
     MPIR_Assert(request_ptr->kind == MPIR_REQUEST_KIND__SEND ||
                 request_ptr->kind == MPIR_REQUEST_KIND__RECV);
 
-    if (request_ptr->kind == MPIR_REQUEST_KIND__SEND) {
-        /* FIXME: are Ibsend requests added to the send queue? */
-        MPII_SENDQ_FORGET(request_ptr);
-    }
-
```

Comment:
Is this (commit "remove send request from debugq when freeing") critical? I think user cares about when a send request completes rather than when it is freed. User always knows when he frees the request in his code.

Reply:
I see. Having separate completion info makes sense. Are we able to support that?

Reply:
Yes, we support it now. `MPIR_Debugq` contains a pointer back to the `MPIR_Request`. We are able to access the completion counter and set a status for the debugger to see based on whether or not `req->cc == 0`.

Reply:
Sounds good. 👍 . Could you point me to the code where we do that?

Reply:
For recv: https://github.com/raffenet/mpich/blob/b229641ec76b562e6ecd7ad3559fcac9da969071/src/mpi/debugger/dll_mpich.c#L577
For send: https://github.com/raffenet/mpich/blob/b229641ec76b562e6ecd7ad3559fcac9da969071/src/mpi/debugger/dll_mpich.c#L726